email_tracking/app.rb

```diff
@@ -0,0 +1,62 @@
+require 'sinatra/base'
+require 'easypost'
+require 'sendgrid-ruby'
+require 'tilt/erb'
+require 'pry'
+require 'dotenv'
+
+include SendGrid
+
+class App < Sinatra::Base
+  set :show_exceptions, :after_handler
+
+  configure :development, :test do
+    Dotenv.load
+  end
+
+  configure do
+    EasyPost.api_key = ENV['EASYPOST_API_KEY']
+    set :sendgrid_key, SendGrid::API.new(api_key: ENV['SENDGRID_API_KEY'])
+  end
+
+
+ post '/easypost-webhook' do
+    sendgrid = settings.sendgrid_key
+    request.body.rewind
+    request_string = request.body.read.to_s
+    response = JSON.parse(request_string)
+
+    if response['object'] == 'Event' && response['description'] == 'tracker.updated'
+      event = EasyPost::Event.receive(request_string)
+      tracker = event.result
+
+      @message = "Hey, this is FunCompany."
+      if tracker.status == 'delivered'
+        @message += "Your package has arrived! "
+      else
+        @message += "There's an update on your package: "
+      end
+
+      tracker.tracking_details.reverse.each do |tracking_detail|
+        if tracking_detail.status == tracker.status
+          @message += "%s says: %s in %s." % [tracker.carrier,tracking_detail.message,tracking_detail.tracking_location.city]
+        end
+        break
+      end
```

Comment:
TrackingDetails come back from our service pre-sorted, so doing a reverse.each will correctly get the most recent detail in the given state. However, as Tribone indicated there can be multiple details for each state so this will create some messy strings.

You should be able to do a `td = tracker.tracking_details.reverse.find{ tracking_detail.status == tracker.status }` and then just build the `message` from that detail